Campaign import export user docs (#361)
mautic/user-documentation: docs/campaigns/exporting_campaigns.rst

Author: This is an exception because we are referring to a Mautic feature in the title.

Exporting Campaigns
###################

> Before importing or exporting data, back up your database as a safety precaution in case you need to restore it. Speak to the administrator of your domain to do this, as it requires specific technical knowledge.

Reviewer: ```suggestion
Before importing or exporting data, back up your database as a safety precaution in case you need to restore it. Speak to the administrator of your Mautic instance to do this, as it requires specific technical knowledge.
```

.. important::

   Both the import and export features are available only in Mautic 7.0 and later versions.

Supported data types
********************

When you select a Campaign, the export feature extracts all Campaign data and entities, along with any dependencies the Campaign needs to function, including:

* Dynamic Content

> * Asset\*

Reviewer: What does this asterisk indicate? I'd like us to include the reference if we're referring to something, right after these bullets.

* Other related dependencies

> 

Reviewer 2: Copilot's suggested footnote ("The exported Campaign includes only directly associated Assets") is accurate and clears up what the asterisk refers to. Apply the suggestion.

=========================

#. Go to the **Campaigns** menu.
#. Click the three-dot icon next to the Campaign's name that you want to export to open the options.
#. Select the **Export** option from the dropdown menu. It automatically downloads the ZIP file.

   |

   .. image:: images/export_campaign.png
      :alt: Highlight of Campaign menu, Options three-dot button, and Export option in the Mautic Campaigns section

   |

.. vale off

2.\ Using the command line
==========================

.. vale on

Use the following commands to save the exported file:

**1. In a specific directory**

.. code-block:: bash
    
    bin/console mautic:entity:export --entity=campaign --id=1 --zip-file --path=path/to-file

**Command parameters**

* ``--entity``: defines the type of entity to export, in this case, ``campaign``.
* ``--id``: defines the ID of the Campaign to export. When you view or edit the Campaign, look at the URL to find the ID. For example, ``/s/campaigns/view/123``, where ``123`` is the ID.
* ``--zip-file``: creates a ZIP file of the Campaign and its dependencies.
* ``--path``: specifies the directory to save the exported file.

**2. In a JSON file**

.. code-block:: bash
    

>     bin/console mautic:entity:export --entity=campaign --id=1 --json-file

Reviewer 2: One technical correction - in the "In a JSON file" example, the flag is --json-only (not --json-file). Verified against EntityExportCommand in core. Otherwise the  CLI/API examples all match the codebase.

* Creates only a JSON file and ignores any additional resources.

3.\ Using Mautic API
====================

You can programmatically export Campaigns using the Mautic API. To do this, you must authenticate your API request with the credentials stored in Mautic's settings.

.. vale off

For more details on how to authenticate, see the :doc:`Authentication page </authentication/authentication>`.

> 

Reviewer 3: ```suggestion

.. important::

   * The API uses a GET request to export a specific Campaign by ID.   
    
   * Before running the following cURL or Python examples, ensure you:

     * Replace ``example.com`` with the actual Mautic instance domain.
     * Replace ``YOUR_ACCESS_TOKEN`` with a valid authentication token.
     * Verify that the User account has the necessary API permissions.

```

Reviewer 3: @vukovicpredrag I think moving the instructions to before the code examples like this would be better for visibility?

Reviewer 2: Yes, above the code examples is the right.

.. vale on

cURL example
------------

.. code-block:: bash

>    curl --location 'https://{your-mautic-domain}/api/campaigns/export/1' \

Reviewer: I wonder if we should make it explicit here that you replace everything in {your-mautic-domain} including the curly braces, or would that be obvious? Also we use that here, and example.com later. Why not use the same?

>    curl --location 'https://{your-mautic-domain}/api/campaigns/export/1' \

Reviewer 3: ```suggestion
   curl --location 'https://example.com/api/campaigns/export/1' \
```

Reviewer 2: Standardizing on example.com works and resolves the inconsistency Ruth flagged. One thing: a bare example.com doesn't signal "replace this" as clearly as a {placeholder} would, so let's make sure the surrounding text tell  the reader to swap example.com for their own Mautic instance domain. With that note added, this looks good.

Reviewer 3: @vukovicpredrag we actually have an "important" admonition to address replacing the placeholder at the bottom:

https://github.com/dcjarvis/mauticuser-documentation/blob/b61e63848885b8e42cb528b0889d742f662ffd2c/docs/campaigns/exporting_campaigns.rst?plain=1#L125-L130 

Now that you mentioned that, this might get missed if folks don't read the full docs until the next example. I'll see what we can do to make this noticable. 👍🏼

Reviewer 2: @adiati98 Moving it above the code blocks (or even repeating a short "replace example.com" note inline) should solve the visibility issue. Either way works for me readers  just need to see it before they copy-paste the curl.

   --header 'Authorization: Bearer YOUR_ACCESS_TOKEN' \

>    --data ''

Reviewer 3: @vukovicpredrag is this review from co-pilot correct? That we should remove the `--data ''`?

>    --data ''

Reviewer 2: Yes, Copilot is right. --data flag forces curl to POST even with an empty body, so dropping --data '' makes it a proper GET (which is what the export endpoint  expects). Apply the suggestion.

Python example
--------------

.. code-block:: python

   import requests

   # API Endpoint
   campaign_id = 1
   url = f'https://example.com/api/campaigns/export/{campaign_id}'

   # Authentication
   headers = {
       'Authorization': 'Bearer YOUR_ACCESS_TOKEN'
   }

   # Send export request
   response = requests.get(url, headers=headers)

   # Handle response
   if response.status_code == 200:
       export_file = response.content
       # Save or process the exported campaign

.. important::

   * Replace ``example.com`` with your actual Mautic instance domain
   * Replace ``YOUR_ACCESS_TOKEN`` with a valid authentication token
   * The API uses a GET request to export a specific Campaign by ID
   * Ensure you have the necessary API permissions

> 

Reviewer 3: ```suggestion
```

Exporting Campaigns
###################

Before importing or exporting data, back up your database as a safety precaution in case you need to restore it. Speak to the administrator of your Mautic instance to do this, as it requires specific technical knowledge.

.. important::

   Both the import and export features are available only in Mautic 7.0 and later versions.

Supported data types
********************

When you select a Campaign, the export feature extracts all Campaign data and entities, along with any dependencies the Campaign needs to function, including:

* Dynamic Content
* Asset\*
* Other related dependencies

.. note::

   \* The exported Campaign includes only directly associated Assets.

> The export command:
> 
> * Detect use of Plugins and Custom Fields
> * Include the data to support these dependencies

Reviewer 3: Add a blank line.

```suggestion

The export command:

* Detect use of Plugins and Custom Fields
* Include the data to support these dependencies
```

.. important::

   * The API uses a GET request to export a specific Campaign by ID.   
    
   * Before running the following cURL or Python examples, ensure you:

     * Replace ``example.com`` with the actual Mautic instance domain.
     * Replace ``YOUR_ACCESS_TOKEN`` with a valid authentication token.
     * Verify that the User account has the necessary API permissions.

.. vale on

cURL example
------------

.. code-block:: bash

   curl --location 'https://example.com/api/campaigns/export/1' \
   --header 'Authorization: Bearer YOUR_ACCESS_TOKEN'

Python example
--------------

.. code-block:: python

   import requests

   # API Endpoint
   campaign_id = 1
   url = f'https://example.com/api/campaigns/export/{campaign_id}'

   # Authentication
   headers = {
       'Authorization': 'Bearer YOUR_ACCESS_TOKEN'
   }

   # Send export request
   response = requests.get(url, headers=headers)

   # Handle response
   if response.status_code == 200:
       export_file = response.content
       # Save or process the exported campaign

.. vale off

How Campaign export works

> *********************************

Reviewer 3: Remove excess asterisks.

```suggestion
How Campaign export works
*************************
```

Exporting Campaigns
###################

Before importing or exporting data, back up your database as a safety precaution in case you need to restore it. Speak to the administrator of your domain to do this, as it requires specific technical knowledge.

.. important::

   Both the import and export features are available only in Mautic 7.0 and later versions.

Supported data types
********************

When you select a Campaign, the export feature extracts all Campaign data and entities, along with any dependencies the Campaign needs to function, including:

* Dynamic Content
* Asset\*
* Custom Fields
* Other related dependencies

The export command:

* Detect use of Plugins and Custom Fields
* Include the data to support these dependencies

.. important::
    

>    The importing instance needs the same Custom Fields and Plugins to be present.

Reviewer 3: I'm not sure if user docs need these info.

But if we do need it, I particularly don't understand about the export command here. What do we actually want to explain?

Reviewer 4: @vukovicpredrag can you validate? In the paragraph above it states that custom fields are part of the extraction and in this sentece that it needs custom fields must be present. I'm unsure which is true.

Reviewer 2: @escopecz @adiati98  Custom Fields are not included in the campaign export. The export only contains references to them. The importing instance needs the same custom fields already present.

Reviewer 3: @vukovicpredrag @escopecz now I'm confused. 😅

Which one should we omit and which to keep?

Reviewer 4: I think Like this is correct for the current state:
<img width="793" height="472" alt="Screenshot 2026-03-06 at 15 59 25" src="https://github.com/user-attachments/assets/5f0160bd-5356-4a6a-a4da-9970b0f34779" />

.. vale off

How the exporting Campaigns works
*********************************

.. vale on

Whether exporting via Mautic instance, the command line, or the API, the process follows these logic steps:

Permissions
    Checks that the logged-in User has the correct permissions to export.

Bulk export
    Supports exporting multiple Campaigns simultaneously.

Data structure
    Exports data in a structured JSON format to ensure compatibility.

Asset management
    Exports Assets into a separate folder in their original format.

File packaging
    Zips the resulting collection of files into a single package for easy transfer across systems.

> .. vale off

Reviewer 3: I'm thinking if we should move this section to after the "Exporting a Campaign" section. Because in my opinion, this is more like a supplement info if users want to know how the feature works under the hood.

Reviewer 2: In my opinion, this section should be moved after the "Exporting a Campaign" section. Practical steps first, technical details second.

.. vale off

2.\ Using the command line
==========================

.. vale on

Use the following commands to save the exported file:

**1. In a specific directory**

.. code-block:: bash
    
    bin/console mautic:entity:export --entity=campaign --id=1 --zip-file --path=path/to-file

**Command parameters**

* ``--entity``: defines the type of entity to export, in this case, ``campaign``.
* ``--id``: defines the ID of the Campaign to export. When you view or edit the Campaign, look at the URL to find the ID. For example, ``/s/campaigns/view/123``, where ``123`` is the ID.
* ``--zip-file``: creates a ZIP file of the Campaign and its dependencies.
* ``--path``: specifies the directory to save the exported file.

**2. In a JSON file**

.. code-block:: bash
    

>     bin/console mautic:entity:export --entity=campaign --id=1 --json-file

Reviewer 3: ```suggestion
    bin/console mautic:entity:export --entity=campaign --id=1 --json-only
```

Use the following commands to save the exported file:

**1. In a specific directory**

.. code-block:: bash
    
    bin/console mautic:entity:export --entity=campaign --id=1 --zip-file --path=path/to-file

**Command parameters**

* ``entity``: defines the type of entity to export, in this case, ``campaign``.
* ``id``: defines the ID of the Campaign to export. When you view or edit the Campaign, look at the URL to find the ID. For example, ``/s/campaigns/view/123``, where ``123`` is the ID.
* ``zip-file``: creates a ZIP file of the Campaign and its dependencies.
* ``path``: specifies the directory to save the exported file.

**2. In a JSON file**

.. code-block:: bash
    
    bin/console mautic:entity:export --entity=campaign --id=1 --json-file

* Creates only a JSON file and ignores any additional resources.

3.\ Using Mautic API
====================

You can programmatically export Campaigns using the Mautic API. To do this, you must authenticate your API request with the credentials stored in Mautic's settings.

.. vale off

For more details on how to authenticate, see the :doc:`Authentication page </authentication/authentication>`.

.. vale on

> cURL example

Reviewer 3: Do we want to use "Curl", "cURL", or "curl"?

Reviewer 2: cURL - that's the official spelling.
